addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3122,6 +3146,55 @@ const SnippetOptionWidget = Widget.extend({
 
         _restoreTransitions();
     },
+    /**
+     * Redirects background calls to a specific CSS property.
+     *
+     * @param {boolean} previewMode - @see this.selectClass
+     * @param {string} widgetValue
+     * @param {Object} params
+     * @returns {Promise|undefined}
+     */
+    selectBackgroundColor: async function (previewMode, widgetValue, params) {
+        if (previewMode === 'reset' && this._backup) {
+            this.$target.css('background-color', this._backup.color);
+            this.$target.css('background-image', this._backup.image);
+            this.$target[0].className = this._backup.classes;
+            this._backup = undefined;
+            return;
+        } else if (previewMode === false) {
+            this._backup = undefined;
+        } else if (previewMode === true && !this.imageBackup) {
+            this._backup = {
+                color: this.$target[0].style['background-color'],
+                image: this.$target[0].style['background-image'],
+                classes: this.$target[0].className,
+            };
+        }
+        params.extraPropertiesToReset = params.extraPropertiesToReset || [];
+        if (params.withGradients) {
+            // gradient and actual color are mutually exclusive
+            const isGradient = widgetValue.includes('-gradient(');
+            if (isGradient) {
+                params.cssProperty = 'background-image';
+                params.extraPropertiesToReset.push('background-color');
+                const parts = backgroundImageCssToParts(this.$target.css('background-image'));
+                parts.gradient = widgetValue;
+                widgetValue = backgroundImagePartsToCss(parts);
+            } else {
+                params.cssProperty = 'background-color';
+                const wasGradient = this.$target.css('background-image').includes('-gradient(');
+                const isColorCombination = weUtils.isColorCombinationName(widgetValue);
+                if (wasGradient && !isColorCombination) {
+                    const parts = backgroundImageCssToParts(this.$target.css('background-image'));
+                    parts.gradient = undefined;
+                    const combined = backgroundImagePartsToCss(parts);
+                    this.$target.css('background-image', combined);
+                    // params.extraPropertiesToReset.push('background-image');
+                }
+            }
+        }
+        await this.selectStyle(previewMode, widgetValue, params);
+    },
```

Comment:
Same as in my other comment but: did you try what we talked about? Because as advertised, this does not work: select a preset, it is not marked as selected anymore.

Instead of one unique data-select-background-color method which tries to understand what it should do, there should be 3:
- data-select-color-combination: only cares about the preset values it receive, if it is not a preset, it does nothing
- data-select-background-color: only cares about color values it receive, ignores the others. Calls select-style with the right params
- data-select-background-gradient: only cares about the gradient values it receive, ignores the others. Call select-style with the right params.

So nicely split methods as their features are also supposed to be independant now (since we can combine the 3) and the main point: there will be 3 calls to `_computeWidgetState` (one for each method), so the colorpicker widget can receive 3 individual values (via `setValue`) and hopefully thus mark the right value as selected in each of its tab (may obviously require some changes but the base code to do all of that is there).